### ISSUE TITLE
BrokenProcessPool when running quickstart (M1 Mac)

### ISSUE THREAD

**Issue by bfhealy** [nasa/dorado-scheduling#89]
I'm encountering errors (on an M1 Mac) when working through the quickstart guide.

Here are the lines I ran to install the software within a conda environment with python 3.10:
```
python /Applications/CPLEX_Studio2211/python/setup.py install
pip install git+https://github.com/nasa/radbelt.git
pip install git+https://github.com/nasa/dorado-scheduling
```
Note that I installed radbelt like this because the default installation had issues with the M1 arm64 architecture. 

After the installation, I cloned this repo and ran:
`dorado-scheduling examples/6.fits -o examples/6.ecsv`

and got these errors:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 289, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/bhealy/miniforge3/envs/dorado-env/bin/dorado-scheduling", line 5, in <module>
Traceback (most recent call last):
  File "<string>", line 1, in <module>
    from dorado.scheduling.scripts.main import main
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/scripts/main.py", line 14, in <module>
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 116, in spawn_main
    from .. import mission as _mission
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/mission.py", line 152, in <module>
    orbit=Spice(
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/orbit/spice.py", line 63, in __init__
    exitcode = _main(fd, parent_sentinel)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 125, in _main
    for filename in download_files_in_parallel(kernels, cache=True):
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/data.py", line 1721, in download_files_in_parallel
    prepare(preparation_data)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 289, in run_path
    combined_paths = ProgressBar.map(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 735, in map
    return _run_module_code(code, init_globals, run_name,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 86, in _run_code
    results = cls.map_unordered(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 843, in map_unordered
    exec(code, run_globals)
  File "/Users/bhealy/miniforge3/envs/dorado-env/bin/dorado-scheduling", line 5, in <module>
    from dorado.scheduling.scripts.main import main
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/scripts/main.py", line 14, in <module>
    for i, f in enumerate(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/_base.py", line 222, in as_completed
    from .. import mission as _mission
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/mission.py", line 152, in <module>
    orbit=Spice(
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/orbit/spice.py", line 63, in __init__
    fs = set(fs)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 844, in <genexpr>
    for filename in download_files_in_parallel(kernels, cache=True):
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/data.py", line 1721, in download_files_in_parallel
    as_completed(p.submit(function, item) for item in items)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 737, in submit
    combined_paths = ProgressBar.map(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 735, in map
    self._adjust_process_count()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 697, in _adjust_process_count
    results = cls.map_unordered(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 843, in map_unordered
    self._spawn_process()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 714, in _spawn_process
    for i, f in enumerate(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/_base.py", line 222, in as_completed
    p.start()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/process.py", line 121, in start
    fs = set(fs)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 844, in <genexpr>
    self._popen = self._Popen(self)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/context.py", line 288, in _Popen
    return Popen(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    as_completed(p.submit(function, item) for item in items)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 737, in submit
    super().__init__(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 154, in get_preparation_data
    self._adjust_process_count()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 697, in _adjust_process_count
    _check_not_importing_main()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
    self._spawn_process()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 714, in _spawn_process
    p.start()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/context.py", line 288, in _Popen
    return Popen(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 289, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 96, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/bhealy/miniforge3/envs/dorado-env/bin/dorado-scheduling", line 5, in <module>
    from dorado.scheduling.scripts.main import main
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/scripts/main.py", line 14, in <module>
    from .. import mission as _mission
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/mission.py", line 152, in <module>
    orbit=Spice(
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/orbit/spice.py", line 63, in __init__
    for filename in download_files_in_parallel(kernels, cache=True):
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/data.py", line 1721, in download_files_in_parallel
    combined_paths = ProgressBar.map(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 735, in map
    results = cls.map_unordered(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 843, in map_unordered
    for i, f in enumerate(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/_base.py", line 222, in as_completed
    fs = set(fs)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 844, in <genexpr>
    as_completed(p.submit(function, item) for item in items)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 737, in submit
    self._adjust_process_count()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 697, in _adjust_process_count
    self._spawn_process()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/process.py", line 714, in _spawn_process
    p.start()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/context.py", line 288, in _Popen
    return Popen(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
|>------------------------------------------------------------------------------------------------|   0 /  3  ( 0.00%)
Traceback (most recent call last):
  File "/Users/bhealy/miniforge3/envs/dorado-env/bin/dorado-scheduling", line 5, in <module>
    from dorado.scheduling.scripts.main import main
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/scripts/main.py", line 14, in <module>
    from .. import mission as _mission
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/mission.py", line 152, in <module>
    orbit=Spice(
  File "/Users/bhealy/dorado-scheduling/dorado/scheduling/orbit/spice.py", line 63, in __init__
    for filename in download_files_in_parallel(kernels, cache=True):
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/data.py", line 1721, in download_files_in_parallel
    combined_paths = ProgressBar.map(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 735, in map
    results = cls.map_unordered(
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/site-packages/astropy/utils/console.py", line 847, in map_unordered
    results.append(f.result())
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/Users/bhealy/miniforge3/envs/dorado-env/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
```

**Comment by lpsinger** [comment]
I haven't seen this before.

**Comment by lpsinger** [comment]
Fixed by #92.
